agent/consul/server.go

```diff
@@ -1237,7 +1247,17 @@ func (s *Server) setupExternalGRPC(config *Config, logger hclog.Logger) {
 	})
 	s.peerStreamServer.Register(s.externalGRPCServer)
 
-	resource.NewServer(resource.Config{}).Register(s.externalGRPCServer)
+	registry := resource.NewRegistry()
+
+	if s.config.DevMode {
+		demo.Register(registry, nil)
+	}
+
+	resourcegrpc.NewServer(resourcegrpc.Config{
+		Registry: registry,
+		Backend:  backend,
+		Logger:   logger.Named("grpc-api.resource"),
```

Comment:
Remember to pass the `ACLResolver` in here 🙇🏻

Reply:
Nice catch! Fixed in https://github.com/hashicorp/consul/pull/16842/commits/04818a63a4e1e069bba34b6a9aa7a99b9b0afa4b